src/odemis/driver/test/technolution_test.py

```diff
@@ -1682,6 +1683,29 @@ def test_assemble_megafield_metadata(self):
             self.assertIsInstance(individual_cell.x_eff_orig, int)
             self.assertIsInstance(individual_cell.y_eff_orig, int)
 
+    def test_calibration_metadata(self):
+        """Test the correct calibration metadata is returned."""
+        original_md = self.MPPC.getMetadata().get(model.MD_CALIB, None)
+
+        # Create the settings observer to store the settings on the metadata
+        settings_obs = SettingsObserver(model.getComponents())
+        all_settings = settings_obs.get_all_settings()
+
+        # Test that if the metadata contains all settings, the required selection
+        # is returned by _get_calibration_metdata()
+        self.MPPC.updateMetadata({model.MD_CALIB: all_settings})
+        calibration_metadata = self.MPPC._get_calibration_metadata()
+        for k in SETTINGS_SELECTION.keys():
+            self.assertTrue(k in calibration_metadata.keys())
+
+        # Test that if the calibration metadata is None, the correct string is returned.
+        self.MPPC.updateMetadata({model.MD_CALIB: None})
+        calibration_metadata = self.MPPC._get_calibration_metadata()
+        self.assertEqual(calibration_metadata,
+                         "Failed getting calibration metadata, something might have gone wrong.")
```

Comment:
Then hopefully the string never changes :D

Reply:
I would hope that is someone updates the code, they also re-run and update the test cases :)